wordpress-mobile/WordPress-Android
# Changed timing of completion of reader QS task.

After the reader improvements project, Reader Quick Start task (Follow Other Sites) became impossible to complete.

I changed the timing of task completion to when you navigate to reader after starting a task.

To test:
- Create new site and start a QuickStart flow. (or use an existing site with active Quick Start).
- Navigate to `Grow Your Audience` Quick Start task list.
- Select `Follow other sites` task.
- When prompted, tap on the Reader bottom in the bottom bar.
- Navigate back to My Site fragment from Reader.
- Confirm that `Grow Your Audience` is completed.

PR submission checklist:
